core/src/main/scala/cats/Show.scala

```diff
@@ -27,7 +32,7 @@ object Show {
 
   final case class Shown(override val toString: String) extends AnyVal
   object Shown {
-    implicit def mat[A](x: A)(implicit z: Show[A]): Shown = Shown(z show x)
+    implicit def mat[A](x: A)(implicit z: ContravariantShow[A]): Shown = Shown(z show x)
```

Comment:
Unfortunately type inference looks to be just slightly shy, like it was when I asked for `[A, B >: A](x: A)(implicit z: Show[B])`: it never converts anything to a `Shown`.

```
[error] /Users/edmund/workspace/scala/cats/tests/src/test/scala/cats/tests/ShowTests.scala:29: type mismatch;
[error]  found   : TimeOfDay
[error]  required: cats.Show.Shown
[error]     assertResult("Good morning, Whiskers!")(show"Good $tod, $cat!")
[error]                                                        ^
[error] /Users/edmund/workspace/scala/cats/tests/src/test/scala/cats/tests/ShowTests.scala:29: type mismatch;
[error]  found   : Cat
[error]  required: cats.Show.Shown
[error]     assertResult("Good morning, Whiskers!")(show"Good $tod, $cat!")
[error]                                                              ^
[error] /Users/edmund/workspace/scala/cats/tests/src/test/scala/cats/tests/ShowTests.scala:31: type mismatch;
[error]  found   : TimeOfDay
[error]  required: cats.Show.Shown
[error]     assertResult("Good morning, Whiskers!")(show"Good $tod, ${List(cat).head}!")
[error]                                                        ^
[error] /Users/edmund/workspace/scala/cats/tests/src/test/scala/cats/tests/ShowTests.scala:31: type mismatch;
[error]  found   : Cat
[error]  required: cats.Show.Shown
[error]     assertResult("Good morning, Whiskers!")(show"Good $tod, ${List(cat).head}!")
[error]                                                                         ^
[error] /Users/edmund/workspace/scala/cats/tests/src/test/scala/cats/tests/ShowTests.scala:46: type mismatch;
[error]  found   : Cat
[error]  required: cats.Show.Shown
[error]     assertResult("Good morning, Whiskers!")(show"Good morning, $cat!")
[error]                                                                 ^
[error] 5 errors found
```